pkg/scheduler/scheduler_test.go

```diff
@@ -345,7 +345,7 @@ func TestSchedulerNoPhantomPodAfterExpire(t *testing.T) {
 
 	waitPodExpireChan := make(chan struct{})
 	timeout := make(chan struct{})
-	errChan := make(chan error)
+	errChan := make(chan error, 1)
```

Comment:
The other three fixes look good now, but this one still seems sloppy. Looking at the code, the whole go routine should probably be replaced with a call to wait.Until or something like that.